GitCommands/PathUtil.cs

```diff
@@ -159,7 +160,93 @@ private static string ResolveRelativePath(string path, string relativePath)
 
         internal static bool IsWslPath(string path)
         {
-            return path.ToLower().StartsWith(@"\\wsl$\");
+            return path.ToLower().StartsWith(WslPrefix);
+        }
+
+        /// <summary>
+        /// Get the name of the distribution (like "Ubuntu-20.04") for WSL2 paths.
+        /// </summary>
+        /// <param name="path">Directory path.</param>
+        /// <returns>Name of the distro or empty for non WSL2 paths</returns>
+        public static string GetWslDistro(string path)
+        {
+            int distroLen = GetWslDistroLength(path);
+            return distroLen <= 0 ? "" : path.Substring(WslPrefix.Length, distroLen);
+
+            static int GetWslDistroLength(string path)
+            {
+                if (string.IsNullOrWhiteSpace(path) || !IsWslPath(path))
+                {
+                    return -1;
+                }
+
+                return path.IndexOfAny(new[] { '\\', '/' }, WslPrefix.Length) - WslPrefix.Length;
+            }
+        }
+
+        /// <summary>
+        /// Convert a path for the Git executable.
+        /// For native (Windows) the path is the same, for WSL Git the path may be transformed.
+        /// </summary>
+        /// <param name="path">Path as seen by the Git Extensions native (Windows) executable.</param>
+        /// <param name="wslDistro">Name of the distro or empty for non WSL paths</param>
+        /// <returns>Posix path if not a WSL distro (Windows Git), WSL path for WSL Git.</returns>
+        public static string GetRepoPath(string? path, string wslDistro)
```

Comment:
❗  The annotation is incorrect: if `path` is nullable - this returns `null` on line:198. 
All callers of this method assume the return isn't null, and subsequently may result in NREs.

Reply:
Changing to not return null